### PR TITLE
WebDriver: Timeouts

### PR DESCRIPTION
**Pull request by linusg**
Ticks off two more boxes in #15551

image

### CODE REVIEW

File: Userland/Services/WebDriver/Client.cpp [hunk 1]
```diff
@@ -442,6 +443,23 @@ ErrorOr<JsonValue, HttpError> Client::handle_get_timeouts(Vector<AK::StringView>
     return make_json_value(timeouts);
 }
 
+// 9.2 Set Timeouts, https://w3c.github.io/webdriver/#dfn-set-timeouts
+// POST /session/{session id}/timeouts
+ErrorOr<JsonValue, HttpError> Client::handle_set_timeouts(Vector<AK::StringView> const& parameters, AK::JsonValue const& payload)
```
Comment (TobyAsE) [inline Userland/Services/WebDriver/Client.cpp:448]: Is there a special reason the contents of this live here? All other handlers forward to their counterparts in `Session`.

File: Userland/Services/WebDriver/Client.cpp [hunk 2]
```diff
@@ -442,6 +443,23 @@ ErrorOr<JsonValue, HttpError> Client::handle_get_timeouts(Vector<AK::StringView>
...
+    auto* session = TRY(find_session_with_id(parameters[0]));
+
+    // 1. Let timeouts be the result of trying to JSON deserialize as a timeouts configuration the request’s parameters.
+    auto timeouts = TRY(json_deserialize_as_a_timeouts_configuration(payload));
+
+    // 2. Make the session timeouts the new timeouts.
+    session->set_timeouts_configuration(timeouts);
+
+    // 3. Return success with data null.
+    return make_json_value(JsonValue {});
```
Comment (TobyAsE) [inline Userland/Services/WebDriver/Client.cpp:460]: I mostly see `JsonValue()`, is there a reason to use `{}` here? whatever we do, it should be consistent...

Comment (linusg) [reply Userland/Services/WebDriver/Client.cpp:460]: https://stackoverflow.com/a/18222927/5952681 - no _reason_, but I'd prefer `{}` generally.

File: Userland/Services/WebDriver/Client.cpp (hunk 1 above)
Comment (linusg) [reply Userland/Services/WebDriver/Client.cpp:448]: Eh, I didn't really see the point but sure :)

**Comment by linusg** [comment]
Fixed, with some additional cleanups :)

**Review by TobyAsE — APPROVED** [review]

**Pull request merged**